src/components/Location.react.js

```diff
@@ -81,23 +84,25 @@ export default class Location extends Component {
         }
     }
 
+    onLocationChange() {
+        const {setProps} = this.props;
+        setProps({
+            pathname: window.location.pathname,
+            href: window.location.href,
+            hash: window.location.hash,
+            search: window.location.search,
+        });
+
+        History.dispatchChangeEvent();
+    }
+
     componentDidMount() {
-        const listener = () => {
-            return () => {
-                const {setProps} = this.props;
-                setProps({
-                    pathname: window.location.pathname,
-                    href: window.location.href,
-                    hash: window.location.hash,
-                    search: window.location.search,
-                });
-            };
-        };
-        window.addEventListener('onpopstate', listener());
-        window.onpopstate = listener('POP');
+        window.onpopstate = this.onLocationChange;
 
-        // non-standard, emitted by Link.react
-        window.addEventListener('onpushstate', listener());
+        window.addEventListener(
+            '_dashprivate_onpushstate',
+            this.onLocationChange
+        );
```

Comment:
Location triggers a `location changed` event on both (1) click on dcc.Link with `refresh=false (default)`, (2) user clicking `back` and `forward` buttons in the browser UI. Prior to this change it was only possible to listen for `(1)` with `"onpushstate"`

Reply:
nice :relaxed: 
the location history feature is working well